lib/beaker/hypervisor/aws_sdk.rb

```diff
@@ -220,125 +221,176 @@ def kill_zombie_volumes
-              :volume_size => host['volume_size'] || rest[:volume_size],
-              # This is required to override the images default for
-              # delete_on_termination, forcing all volumes to be deleted once the
-              # instance is terminated.
-              :delete_on_termination => true,
-            }
+      # Transform the images block_device_mappings output into a format
+      # ready for a create.
+      orig_bdm = image.block_device_mappings()
+      @logger.notify("aws-sdk: Image block_device_mappings: #{orig_bdm.to_hash}")
+      block_device_mappings = []
+      orig_bdm.each do |device_name, rest|
+        block_device_mappings << {
+          :device_name => device_name,
+          :ebs => {
+            # Change the default size of the root volume.
+            :volume_size => host['volume_size'] || rest[:volume_size],
+            # This is required to override the images default for
+            # delete_on_termination, forcing all volumes to be deleted once the
+            # instance is terminated.
+            :delete_on_termination => true,
           }
-        end
-
-        security_group = ensure_group(vpc || region, Beaker::EC2Helper.amiports(host))
-
-        # Launch the node, filling in the blanks from previous work.
-        @logger.notify("aws-sdk: Launch instance")
-        config = {
-          :count => 1,
-          :image_id => image_id,
-          :monitoring_enabled => true,
-          :key_pair => ensure_key_pair(region),
-          :security_groups => [security_group],
-          :instance_type => amisize,
-          :disable_api_termination => false,
-          :instance_initiated_shutdown_behavior => "terminate",
-          :block_device_mappings => block_device_mappings,
-          :subnet => subnet_id,
         }
-        instance = region.instances.create(config)
+      end
 
-        # Persist the instance object for this host, so later it can be
-        # manipulated by 'cleanup' for example.
-        host['instance'] = instance
+      security_group = ensure_group(vpc || region, Beaker::EC2Helper.amiports(host))
 
-        @logger.notify("aws-sdk: Launched #{host.name} (#{amitype}:#{amisize}) using snapshot/image_type #{image_type}")
-      end
+      # Launch the node, filling in the blanks from previous work.
+      @logger.notify("aws-sdk: Launch instance")
+      config = {
+        :count => 1,
+        :image_id => image_id,
+        :monitoring_enabled => true,
+        :key_pair => ensure_key_pair(region),
+        :security_groups => [security_group],
+        :instance_type => amisize,
+        :disable_api_termination => false,
+        :instance_initiated_shutdown_behavior => "terminate",
+        :block_device_mappings => block_device_mappings,
+        :subnet => subnet_id,
+      }
+
+      {instance: region.instances.create(config),
+       amitype: amitype,
+       amisize: amisize,
+       image_type: image_type}
```

Comment:
@kbarber pointed out that this probably isn't idiomatic formatting, so I'll change the brace placement, indentation, etc. in the next version.  He also mentioned that the "foo:" form may not be supported by all the relevant versions of Ruby (so I'll see about that as well).